x-pack/plugins/task_manager/server/task_poller.ts

```diff
@@ -89,11 +92,15 @@ export function createTaskPoller<T, H>({
     concatMap(async (set: Set<T>) => {
       closeSleepPerf();
       return mapResult<H, Error, Result<H, PollingError<T>>>(
-        await promiseResult<H, Error>(work(...pullFromSet(set, getCapacity()))),
+        await promiseResult<H, Error>(
+          timeoutPromiseAfter<H, Error>(
+            work(...pullFromSet(set, getCapacity())),
+            workTimeout ?? pollInterval,
```

Comment:
What is the case for `workTimeout` being `undefined`?  Tests?  I worry that if there a path in production code to have this as `undefined`, the `pollInterval` value used would be too close to a valid cycle - should it at least be some kind of multiple of `pollInterval`, maybe like 2 or something?

Reply:
hmm it was really just a default... but yeah, you're right that if someone sets the interval to be lower than the latency of talking to Elasticsearch this will behave badly 🤔 
That said... same is true for the actual polling....

Not sure about the `* 2` approach, as we have no idea what the interval will be and it might be set to something ridiculous like 30m, in which case, timeout will be 60m....
That said... if your poll is 30m you don't care about immediate results anyway, do you 🤔

Reply:
As the default config is set to `10` I think we can leave it as is for now and see how it behaves.
Unless someone specifically changes it, it'll always be far longer than the interval itself, and as this is here as a safeguard against issues in marking tasks as running (this time *does not* include task execution) I feel comfortable waiting to see how it behaves in the real world.